codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java

```diff
@@ -65,15 +65,17 @@ public static boolean isAsteriskForm(URI uri) {
      * {@link HttpVersion#isKeepAliveDefault()}.
      */
     public static boolean isKeepAlive(HttpMessage message) {
-        CharSequence connection = message.headers().get(HttpHeaderNames.CONNECTION);
-        if (HttpHeaderValues.CLOSE.contentEqualsIgnoreCase(connection)) {
+        boolean connectionClose = message.headers().containsValue(HttpHeaderNames.CONNECTION,
+                HttpHeaderValues.CLOSE, true);
+        if (connectionClose) {
             return false;
         }
 
         if (message.protocolVersion().isKeepAliveDefault()) {
-            return !HttpHeaderValues.CLOSE.contentEqualsIgnoreCase(connection);
+            return !connectionClose;
```

Comment:
connectionClose can only be false here

Reply:
@violetagg ^^ can you adjust this  ... @slandelle good catch!

Reply:
done